### PR TITLE
UIREC-294 Check if a holding exists during the abandonment check

### PR DESCRIPTION
**Pull request by usavkov-epam**
## Purpose

https://issues.folio.org/browse/UIREC-294

When a holding of a piece changes, a check is made to see if the current holding has connections with other items and/or pieces. The absence of links to the current holding implies a proposal to the user to delete or keep this holding. However, it may already have been deleted (as evidenced by the "invalid reference" label), so making additional checks to ensure that it will be abandoned is meaningless.

## Approach

Check if a holding exists before analyzing if it will be abandoned after changing a piece holding.

## Screenshots

https://github.com/folio-org/ui-receiving/assets/88109087/b38e8d2b-9396-4191-85ac-ca8ca41006c9

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/TitleDetails/AddPieceModal/AddPieceModal.test.js [hunk 1]
```diff
@@ -53,11 +64,21 @@ const renderAddPieceModal = (props = defaultProps) => (render(
   { wrapper: MemoryRouter },
 ));
 
+const findButton = (name) => screen.findByRole('button', { name });
+
+const createNewHoldingForThePiece = (newHoldingId = 'newHoldingUUID') => {
+  return act(async () => FieldInventory.mock.calls[0][0].onChange(null, 'locationId', 'holdingId', newHoldingId));
+}
```
Comment (Dmytro-Melnyshyn) [inline src/TitleDetails/AddPieceModal/AddPieceModal.test.js:71]: please fix eslint warnings

Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.test.js]: Done, thanks!

File: package.json [hunk 2]
```diff
@@ -218,9 +218,9 @@
     "react-router-prop-types": "^1.0.4"
   },
   "optionalDependencies": {
-    "@folio/plugin-find-instance": "^7.0.0",
-    "@folio/plugin-find-organization": "^5.0.0",
-    "@folio/plugin-find-po-line": "^5.0.0"
+    "@folio/plugin-find-instance": "^6.3.0",
+    "@folio/plugin-find-organization": "^4.0.0",
+    "@folio/plugin-find-po-line": "^4.0.0"
```
Comment (alisher-epam) [inline package.json:223]: Sorry. Why did we downgrade packages?

File: src/TitleDetails/AddPieceModal/AddPieceModal.test.js [hunk 3]
```diff
@@ -1,10 +1,13 @@
-import React from 'react';
 import { MemoryRouter } from 'react-router-dom';
-import { render, screen } from '@folio/jest-config-stripes/testing-library/react';
+
 import user from '@folio/jest-config-stripes/testing-library/user-event';
+import { act, render, screen } from '@folio/jest-config-stripes/testing-library/react';
```
Comment (alisher-epam) [inline src/TitleDetails/AddPieceModal/AddPieceModal.test.js:5]: Should remove the line

File: src/TitleDetails/AddPieceModal/AddPieceModal.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ import {
   TextField,
   checkScope,
 } from '@folio/stripes/components';
+import { useOkapiKy } from '@folio/stripes/core';
 import stripesFinalForm from '@folio/stripes/final-form';
 import {
```
Comment (alisher-epam) [inline src/TitleDetails/AddPieceModal/AddPieceModal.js:22]: add extra empty line please

Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.js:22]: why?

File: package.json (hunk 2 above)
Comment (usavkov-epam) [reply package.json:223]: Oh, we didn't, it seems that was applied after rebase, I'll fix it

File: src/TitleDetails/AddPieceModal/AddPieceModal.test.js (hunk 3 above)
Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.test.js]: Done

File: src/TitleDetails/AddPieceModal/AddPieceModal.js (hunk 4 above)
Comment (alisher-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.js:22]: @folio and local imports will be identified easily.
Screenshot 2023-10-12 at 2 16 41 PM

Comment (usavkov-epam) [reply src/TitleDetails/AddPieceModal/AddPieceModal.js:22]: there is already an empty line between folio and local imports, you're showing a collapsed list :)

**Review by Dmytro-Melnyshyn — APPROVED** [review]

**Pull request merged**